acala/src/mappings/mappingHandlers.ts

```diff
@@ -35,35 +35,45 @@ export async function handleBlock(block: SubstrateBlock): Promise<void> {
       handleEvent(block.block.header.number.toString(), idx, evt)
     );
 
-  const evmLogs = await Promise.all(
-    wrappedEvents
-      .filter(evt => {
-        const baseFilter = acalaProcessor.handlerProcessors['substrate/AcalaEvmEvent'].baseFilter[0];
-        return evt.event.section === baseFilter.module && evt.event.method === baseFilter.method;
-      })
-      .map(evt => handleEvmLog(block.block.header.number.toString(), evt))
-  );
+  const evmLogs: EvmLog[][] = []
+  for (const evt of wrappedEvents.filter(evt => {
+    const baseFilter = acalaProcessor.handlerProcessors['substrate/AcalaEvmEvent'].baseFilter[0];
+    return evt.event.section === baseFilter.module && evt.event.method === baseFilter.method;
+  })) {
+    const logResult = await handleEvmLog(block.block.header.number.toString(), evt);
+    evmLogs.push(logResult);
```

Comment:
Having push will make this non-deterministic because there is still an await. There is also nothing awaiting all of these promises.

I think before it was correct and this will lead to more issues.